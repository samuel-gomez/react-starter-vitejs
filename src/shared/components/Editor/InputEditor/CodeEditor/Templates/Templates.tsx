import Button from '@axa-fr/react-toolkit-button/dist/esm/index';
import HelpInfo from 'shared/components/HelpInfo';
import type { TReturnUseCodeEditor } from '../CodeEditor';

const DEFAULT_LIST = [
  {
    id: 'add-list',
    icon: 'list',
    tooltipLabel: 'Liste',
  },
  {
    id: 'add-title__h1',
    label: 'H1',
    tooltipLabel: 'Titre h1',
  },
  {
    id: 'add-title__h2',
    label: 'H2',
    tooltipLabel: 'Titre h2',
  },
  {
    id: 'add-title__h3',
    label: 'H3',
    tooltipLabel: 'Titre h3',
  },
  {
    id: 'add-title__h4',
    label: 'H4',
    tooltipLabel: 'Titre h4',
  },
  {
    id: 'add-text',
    label: 'P',
    tooltipLabel: 'Paragraph',
  },
  {
    id: 'add-restitution__twoColumns',
    icon: 'list-alt',
    tooltipLabel: 'Restitution two columns',
  },
  {
    id: 'add-restitution__oneColumn',
    icon: 'list-alt',
    tooltipLabel: 'Restitution one column',
  },
  {
    id: 'add-modal',
    icon: 'modal-window',
    tooltipLabel: 'Modal',
  },
  {
    id: 'add-panel__basic',
    icon: 'credit-card',
    tooltipLabel: 'Panel basic',
  },
];

type TlistElement = {
  id: string;
  label?: string;
  icon?: string;
  tooltipLabel?: string;
};
export type Tlistelements = TlistElement[];

type TTemplates = {
  list?: Tlistelements;
  submitTemplate: TReturnUseCodeEditor['submitTemplate'];
  onClearCodeEditor: TReturnUseCodeEditor['onClearCodeEditor'];
};

type TonClick = (event: React.MouseEvent<HTMLElement>) => void;

const Templates = ({ list = DEFAULT_LIST, submitTemplate, onClearCodeEditor }: TTemplates) => (
  <div className="af-templates">
    <HelpInfo key="delete" content="Supprimer" classModifier="editor">
      <Button aria-label="Suppression du code" className="af-btn--circle af-btn--delete" onClick={onClearCodeEditor}>
        <i role="img" aria-label="Icone suppression du code" className="glyphicon glyphicon-trash" />
      </Button>
    </HelpInfo>
    {list.map(({ id, label, icon, tooltipLabel }) => (
      <HelpInfo key={id} content={tooltipLabel} classModifier="editor">
        <Button aria-label={`Ajout du code ${label}`} className="af-btn--circle" id={id} onClick={submitTemplate as TonClick}>
          {!!icon && <i role="img" aria-label={`Icone ajout du code ${label}`} className={`glyphicon glyphicon-${icon}`} />}
          {label}
        </Button>
      </HelpInfo>
    ))}
  </div>
);

export default Templates;
