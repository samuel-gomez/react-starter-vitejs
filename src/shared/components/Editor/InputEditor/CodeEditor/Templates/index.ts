export { default } from './Templates';
export type { TlistElements } from './Templates';
