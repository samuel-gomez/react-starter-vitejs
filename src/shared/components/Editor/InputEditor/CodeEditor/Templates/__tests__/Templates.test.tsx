import { describe, it, expect, vi } from 'vitest';
import { render } from '@testing-library/react';
import Templates from '..';

const defaultProps = {
  submitTemplate: vi.fn(),
  onClearCodeEditor: vi.fn(),
};

describe('<Templates/>', () => {
  it('Should render Templates', () => {
    const { asFragment } = render(<Templates {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
