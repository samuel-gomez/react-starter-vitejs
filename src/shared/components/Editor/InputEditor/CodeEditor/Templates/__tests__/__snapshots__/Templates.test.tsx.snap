// Vitest Snapshot v1

exports[`<Templates/> > Should render Templates 1`] = `
<DocumentFragment>
  <div
    class="af-templates"
  >
    <div
      class="af-popover__wrapper"
    >
      <div
        class="af-popover__container af-popover__container--editor"
      >
        <div
          class="af-popover__container-over"
          role="presentation"
        >
          <button
            aria-label="Suppression du code"
            class="af-btn--circle af-btn--delete"
            type="button"
          >
            <i
              aria-label="Icone suppression du code"
              class="glyphicon glyphicon-trash"
              role="img"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="af-popover__wrapper"
    >
      <div
        class="af-popover__container af-popover__container--editor"
      >
        <div
          class="af-popover__container-over"
          role="presentation"
        >
          <button
            aria-label="Ajout du code undefined"
            class="af-btn--circle"
            id="add-list"
            type="button"
          >
            <i
              aria-label="Icone ajout du code undefined"
              class="glyphicon glyphicon-list"
              role="img"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="af-popover__wrapper"
    >
      <div
        class="af-popover__container af-popover__container--editor"
      >
        <div
          class="af-popover__container-over"
          role="presentation"
        >
          <button
            aria-label="Ajout du code H1"
            class="af-btn--circle"
            id="add-title__h1"
            type="button"
          >
            H1
          </button>
        </div>
      </div>
    </div>
    <div
      class="af-popover__wrapper"
    >
      <div
        class="af-popover__container af-popover__container--editor"
      >
        <div
          class="af-popover__container-over"
          role="presentation"
        >
          <button
            aria-label="Ajout du code H2"
            class="af-btn--circle"
            id="add-title__h2"
            type="button"
          >
            H2
          </button>
        </div>
      </div>
    </div>
    <div
      class="af-popover__wrapper"
    >
      <div
        class="af-popover__container af-popover__container--editor"
      >
        <div
          class="af-popover__container-over"
          role="presentation"
        >
          <button
            aria-label="Ajout du code H3"
            class="af-btn--circle"
            id="add-title__h3"
            type="button"
          >
            H3
          </button>
        </div>
      </div>
    </div>
    <div
      class="af-popover__wrapper"
    >
      <div
        class="af-popover__container af-popover__container--editor"
      >
        <div
          class="af-popover__container-over"
          role="presentation"
        >
          <button
            aria-label="Ajout du code H4"
            class="af-btn--circle"
            id="add-title__h4"
            type="button"
          >
            H4
          </button>
        </div>
      </div>
    </div>
    <div
      class="af-popover__wrapper"
    >
      <div
        class="af-popover__container af-popover__container--editor"
      >
        <div
          class="af-popover__container-over"
          role="presentation"
        >
          <button
            aria-label="Ajout du code P"
            class="af-btn--circle"
            id="add-text"
            type="button"
          >
            P
          </button>
        </div>
      </div>
    </div>
    <div
      class="af-popover__wrapper"
    >
      <div
        class="af-popover__container af-popover__container--editor"
      >
        <div
          class="af-popover__container-over"
          role="presentation"
        >
          <button
            aria-label="Ajout du code undefined"
            class="af-btn--circle"
            id="add-restitution__twoColumns"
            type="button"
          >
            <i
              aria-label="Icone ajout du code undefined"
              class="glyphicon glyphicon-list-alt"
              role="img"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="af-popover__wrapper"
    >
      <div
        class="af-popover__container af-popover__container--editor"
      >
        <div
          class="af-popover__container-over"
          role="presentation"
        >
          <button
            aria-label="Ajout du code undefined"
            class="af-btn--circle"
            id="add-restitution__oneColumn"
            type="button"
          >
            <i
              aria-label="Icone ajout du code undefined"
              class="glyphicon glyphicon-list-alt"
              role="img"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="af-popover__wrapper"
    >
      <div
        class="af-popover__container af-popover__container--editor"
      >
        <div
          class="af-popover__container-over"
          role="presentation"
        >
          <button
            aria-label="Ajout du code undefined"
            class="af-btn--circle"
            id="add-modal"
            type="button"
          >
            <i
              aria-label="Icone ajout du code undefined"
              class="glyphicon glyphicon-modal-window"
              role="img"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="af-popover__wrapper"
    >
      <div
        class="af-popover__container af-popover__container--editor"
      >
        <div
          class="af-popover__container-over"
          role="presentation"
        >
          <button
            aria-label="Ajout du code undefined"
            class="af-btn--circle"
            id="add-panel__basic"
            type="button"
          >
            <i
              aria-label="Icone ajout du code undefined"
              class="glyphicon glyphicon-credit-card"
              role="img"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
